src/classes/ADDR_ProcessControl.cls

```diff
@@ -0,0 +1,100 @@
+    * @description Flag to prevent recursion into the Address trigger handler/logic
+    */
+    private static Integer recursionFlag = 0;
+
+    /*******************************************************************************************************
+    * @description Increment/Decrement the static flag counter that tracks whether the Address processing
+    *    trigger has been called already
+    * @param b A boolean where true increments the counter and false decrements it.
+    * @return void This method modifies class variables and doesn't return anything.
+    ********************************************************************************************************/
+    public static void setRecursionFlag(boolean b) {
+        recursionFlag += (b ? 1 : -1);
+        if (recursionFlag < 0) {
+            recursionFlag = 0;
+        }
+        UTIL_Debug.debug('>>>>>> setRecursionFlag(' + b + ') ==> ' + recursionFlag);
+    }
+
+    /* @description Returns true if the recursionFlag value is greater than 1 */
+    public static Boolean checkRecursionState() {
+        UTIL_Debug.debug('>>>>>> checkRecursionState() ==> ' + recursionFlag);
+        return (recursionFlag > 0);
+    }
+
+    /*******************************************************************************************************
+    * @description Simple static flag to prevent execution of the address validation triggers
+    */
+    private static Integer validationRecursionFlag = 0;
+    public static Boolean doNotExecuteValidationTrigger;
+
+    /* @description Returns true if the validationRecursionFlag value is greater than 1 */
+    public static Boolean checkValidationRecursionState() {
+        return (validationRecursionFlag > 0);
+    }
+
+    /*******************************************************************************************************
+    * @description Increment/Decrement the static flag counter used to prevent Address Validation recursion
+    * @param b A boolean where true increments the counter and false decrements it.
+    * @return void This method modifies class variables and doesn't return anything.
+    ********************************************************************************************************/
+    public static void setValidationRecursionFlag(boolean b) {
+        validationRecursionFlag += (b ? 1 : -1);
+        if (validationRecursionFlag < 0) {
+            validationRecursionFlag = 0;
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description To support DML operations with more than 200 records, use a static Set of Id's that
+    * have been passed to the verifyAddressRecords() method. This allows for a recursion check that goes
+    * by the Id's of the records as opposed to a simple boolean.
+    */
+    public static Set<Id> processedIdsForValidation = new Set<Id>();
```

Comment:
@davidhabib  The goal was to prevent the same address from being submitted to the validation more than once. If I'm not mistaken this is necessary because DML does not occur in the validator class as it would in our other triggers. Instead, it queues up an @future method to handle the validation.  In the event that more than 200 records are committed at once, the validator class would need to know which of those records were previously queued up for validation as well as handling general recursion into the validator class. Incrementing a var before calling the @future method and decrementing afterwards wouldn't help with recursion on the validation logic. I'll update the comments around this to clarify.